File: Samples/BasicApplication/BasicApplication/BasicApplication.Web/Modules/Membership/Account/LoginForm.cs

namespace BasicApplication.Membership.Forms
{
    using System;
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System.Collections.Generic;
    using System.ComponentModel;

    [FormScript("Membership.Login")]
    [BasedOnRow(typeof(Administration.Entities.UserRow))]
    public class LoginForm
    {
        [Placeholder("default username is 'admin'")]
        public String Username { get; set; }
        [PasswordEditor, Placeholder("default password for admin user is 'serenity'"), Required(true)]
        public String Password { get; set; }
    }
}

File: serene/src/Serene.Web/Modules/ServerTypes/Membership/LoginForm.ts
import { initFormType, PasswordEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface LoginForm {
    Username: StringEditor;
    Password: PasswordEditor;
}

export class LoginForm extends PrefixedContext {
    static readonly formKey = 'Membership.Login';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!LoginForm.init) {
            LoginForm.init = true;

            initFormType(LoginForm, [
                'Username', StringEditor,
                'Password', PasswordEditor
            ]);
        }
    }
}